app/helpers/language_helper.py

```diff
@@ -13,17 +14,18 @@
 }
 
 
-def handle_language() -> None:
+def handle_language(metadata: Optional[Mapping[str, Any]] = None) -> None:
+    metadata = metadata or get_metadata(current_user)
```

Comment:
This should be inside the `if session` check as it is only needed if that is true.